docs/specs.md

```diff
@@ -127,17 +163,66 @@ Note: the Google Drive folder contains
 
 #### Tabular source document
 
-TBD
-
-#### Free-form contextual source document
-
-This is a lower-level document model in which the document is divided into independent chunks, each one with three elements:
+A Tabular document is one in which the structure has two dimensions, i.e. it is
+organized mainly as rows and columns (so it can be mapped to a table), which
+then contain some type of data. Its semantic premise is that there exists
+some kind of relationships along rows and along colums, relationships that may
+have implications in terms of PII detection and required processing
+approaches. Examples of tabular documents are Excel files, or CSV files.
+
+One particular feature of certain tabular documents is that they can be very
+large, since they may contain huge quantities of data; hence there should be a
+way in which they can be processed by chunks (i.e. as streaming objects),
+without the need to hold all their contents at once.
+
+As all source document model types, a tabular document is considered as a
+document header, plus a sequence of document chunks.
+
+The document header corresponds to the specification in the previous section,
+with one modification in the `context` element: in addition to the already
+defined `document` and `dataset` contexts, there may also be a `column`
+context, providing specific information for each of the columns in the document
+(considered in sequential order). It could contain up to two subelements:
+* `name`: the column names, as a list of text strings
+* `description`: a description for each column, also as a list of text strings
+
+Beyond the header, each chunk contains a small dictionary with 2 or 3 elements:
+* `id`: an arbitrary string that should be unique per document. Its mission is
+  to make it easier later on to map detected PII instances to the chunk they
+  are part of
+* `table`: a 2D array holding the tabular data for this hunk, as a 2-level
+  list of first rows and, inside each row, columns. The contents of each table 
+  position (a table cell) is an arbitrary text string
+* `context`: elements that help to provide additional information to help in
+  the detection and decision process. There can be three types of context, all
+  of them optional:
+   - _global_: it will be a link/transposition of the context in the document
+     header, as described above
+   - _before_ (optional): a small section of data from the chunk preceding this
+     one, containing the closest rows to this chunk
+   - _after_: a small section of data from the chunk following this one,
```

Comment:
Also `optional`?

Reply:
good catch, actually _all_ three fields are optional, as the paragraph before says, but an "(optional)" qualifier was left in _before_, and that's confusing